backend/src/Types/Api.hs

```diff
@@ -27,10 +28,37 @@ data GameReport = GameReport
   }
   deriving (Generic)
 
-instance FromJSON GameReport
+instance FromJSON RawGameReport
+
+instance ToJSON RawGameReport
+
+toGameReport :: UTCTime -> PlayerId -> PlayerId -> RawGameReport -> GameReport
+toGameReport timestamp winnerId loserId r =
+  GameReport
+    { gameReportTimestamp = timestamp,
+      gameReportWinnerId = winnerId,
+      gameReportLoserId = loserId,
+      gameReportSide = r.side,
+      gameReportVictory = r.victory,
+      gameReportMatch = r.match,
+      gameReportCompetition = r.competition,
+      gameReportLeague = r.league,
+      gameReportExpansions = r.expansions,
+      gameReportTreebeard = r.treebeard,
+      gameReportActionTokens = r.actionTokens,
+      gameReportDwarvenRings = r.dwarvenRings,
+      gameReportTurns = r.turns,
+      gameReportCorruption = r.corruption,
+      gameReportMordor = r.mordor,
+      gameReportInitialEyes = r.initialEyes,
+      gameReportAragornTurn = r.aragornTurn,
+      gameReportStrongholds = r.strongholds,
+      gameReportInterestRating = r.interestRating,
+      gameReportComments = r.comments
+    }
 
 data SubmitGameReportResponse = SubmitGameReportResponse
```

Comment:
Do you need anything from this API response other than what's here?

Reply:
I don't think so, maybe a future nice-to-have would be the `pid` of any new players created? If there were to be some kind of profile page for a player, that would let us display a link to it after the submission succeeds. Not needed right now, just thinking out loud

Reply:
Great idea - easily done! I'll do that ~sometime in the future